epo_ops/api.py

```diff
@@ -35,13 +36,30 @@ def __init__(self, key, secret, accept_type="xml", middlewares=None):
         self.secret = secret
         self._access_token = None
 
-    def family(self, reference_type, input, endpoint=None, constituents=None):
+    def family(
+        self,
+        reference_type: str,
+        input: Union[Docdb, Epodoc],
+        constituents: Optional[List[str]] = None,
+    ):
```

Comment:
> Acc. to ops specs, endpoint is not used in family service. Not used in tests either. Options: either document it in docstring or remove it. Went for second option.

Removing parameters means a breaking change, so we would need to target this for a 5.x release. Is it possible to move on with type hinting without doing that, and saving it for a subsequent iteration?

Reply:
In order to prepare the removal, we can issue a deprecation warning when it is used, like:
```python
if endpoint is not None:
    warnings.warn("The `endpoint` argument is not used in this context and will be removed.", DeprecationWarning)
```

Reply:
You are quite right, I was a bit too fast on this one. Thanks.